[dtensor] switch softmax backward ops to OpStrategy

tianyu-l opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119256
* __->__ #119255

As titled. This is a followup to PR #117723 on softmax forward ops.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @wconstab @yf225

tianyu-l committed:
[dtensor] switch softmax backward ops to OpStrategy

[ghstack-poisoned]

tianyu-l committed:
Update on "[dtensor] switch softmax backward ops to OpStrategy"


As titled. This is a followup to PR #117723 on softmax forward ops.

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 wconstab yf225

[ghstack-poisoned]

XilunWu reviewed (approved):
LGTM!

wanchaol reviewed (approved):
Looks good! some minor comments

tianyu-l committed:
Update on "[dtensor] switch softmax backward ops to OpStrategy"


As titled. This is a followup to PR #117723 on softmax forward ops.

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 wconstab yf225

[ghstack-poisoned]
